lxd/db/cluster/cluster_groups.go

```diff
@@ -51,3 +57,43 @@ func (c *ClusterGroup) ToAPI() (*api.ClusterGroup, error) {
 
 	return &result, nil
 }
+
+// GetClusterGroupUsedBy collates references to the cluster group with the given name.
+// This currently only returns the URLs of projects whose `restricted.cluster.groups` configuration
+// contains the cluster group.
+func GetClusterGroupUsedBy(ctx context.Context, tx *sql.Tx, groupName string) ([]string, error) {
+	q := `
+SELECT projects.name, projects_config.value FROM projects 
+JOIN projects_config ON projects.id = projects_config.project_id 
+WHERE projects_config.key = 'restricted.cluster.groups' AND projects_config.value LIKE '%` + groupName + `%'`
```

Comment:
This is a SQL injection security risk.

Reply:
Also the use of LIKE here seems potentially incorrect as it might match partial group names.

Can we use JSON collation and direct matching with query parameters instead?

Reply:
Unfortunately not, because the config key is a comma separated list.

Rather than doing anything clever, I'm now just querying for all projects that have the `restricted.cluster.groups` configuration key set. Then parsing the output to see if it contains the group.

Reply:
makes sense because they may be spaces in the comma delimited list too